File: samples/client/3_1_0_unit_test/java/src/test/java/unit_test_api/components/schemas/RequiredWithEscapedCharactersTest.java
package unit_test_api.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import unit_test_api.configurations.JsonSchemaKeywordFlags;
import unit_test_api.configurations.SchemaConfiguration;
import unit_test_api.exceptions.ValidationException;
import unit_test_api.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class RequiredWithEscapedCharactersTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(new JsonSchemaKeywordFlags.Builder().format().build());

    @Test
    public void testObjectWithSomePropertiesMissingIsInvalidFails() {
        // object with some properties missing is invalid
        final var schema = RequiredWithEscapedCharacters.RequiredWithEscapedCharacters1.getInstance();
        try {
            schema.validate(
                MapUtils.makeMap(
                    new AbstractMap.SimpleEntry<String, String>(
                        "foo\nbar",
                        "1"
                    ),
                    new AbstractMap.SimpleEntry<String, String>(
                        "foo\"bar",
                        "1"
                    )
                ),
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testObjectWithAllPropertiesPresentIsValidPasses() throws ValidationException {
        // object with all properties present is valid
        final var schema = RequiredWithEscapedCharacters.RequiredWithEscapedCharacters1.getInstance();
        schema.validate(
            MapUtils.makeMap(
                new AbstractMap.SimpleEntry<String, Number>(
                    "foo\nbar",
                    1
                ),
                new AbstractMap.SimpleEntry<String, Number>(
                    "foo\"bar",
                    1
                ),
                new AbstractMap.SimpleEntry<String, Number>(
                    "foo\\bar",
                    1
                ),
                new AbstractMap.SimpleEntry<String, Number>(
                    "foo\rbar",
                    1
                ),
                new AbstractMap.SimpleEntry<String, Number>(
                    "foo\tbar",
                    1
                ),
                new AbstractMap.SimpleEntry<String, Number>(
                    "foo\fbar",
                    1
                )
            ),
            configuration
        );
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/RequiredWithEscapedCharactersTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class RequiredWithEscapedCharactersTest {
    companion object {
        val configuration = SchemaConfiguration(JsonSchemaKeywordFlags.Builder().format().build())
    }

    @Test
    fun testObjectWithSomePropertiesMissingIsInvalidFails() {
        // object with some properties missing is invalid
        val schema = RequiredWithEscapedCharacters.RequiredWithEscapedCharacters1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    mapOf(
                        Pair(
                            "foo\nbar",
                            "1"
                        ),
                        Pair(
                            "foo\"bar",
                            "1"
                        )
                    ),
                    configuration
                )
            }
        )
    }

    @Test
    fun testObjectWithAllPropertiesPresentIsValidPasses() {
        // object with all properties present is valid
        val schema = RequiredWithEscapedCharacters.RequiredWithEscapedCharacters1.getInstance()
        schema.validate(
            mapOf(
                Pair(
                    "foo\nbar",
                    1
                ),
                Pair(
                    "foo\"bar",
                    1
                ),
                Pair(
                    "foo\\bar",
                    1
                ),
                Pair(
                    "foo\rbar",
                    1
                ),
                Pair(
                    "foo\tbar",
                    1
                ),
                Pair(
                    "foo\u000Cbar",
                    1
                )
            ),
            configuration
        )
    }
}
